src/main/java/Duke.java

```diff
@@ -1,10 +1,117 @@
+                    continue;
+                }
+                String tl = "";
+                for (int i = 0; i < taskList.size(); i++) {
+                    Task t = taskList.get(i);
+                    tl += (i + 1) + ". " + t.toString() + System.lineSeparator();
+                }
+                printOutput(tl, false);
+            } else if (userInput.toLowerCase().startsWith("done ")) {
+                String taskStr = userInput.substring(4).strip();
+                try {
+                    Task t = taskList.get(Integer.parseInt(taskStr) - 1);
+                    t.setDone(true);
+                    printOutput("Nice! I've marked this task as done:\n\t" + t.toString(), false);
+                } catch (Exception e) {
+                    printOutput("Oops! That didn't work, please check your input and try again!", true);
+                }
+            } else if (userInput.toLowerCase().startsWith("deadline")) {
+                if (!userInput.toLowerCase().contains("/by")) {
+                    printOutput("usage: deadline <name of item> /by <dd-MMM-yyyy HH:mm:ss OR next [weekday]>", true);
+                    continue;
+                }
+                String[] taskArgs = userInput.substring(8).strip().split("/by");
+
+                Deadline d = new Deadline(taskArgs[0].strip(), taskArgs[1].strip());
+                if (d.getDue() == null) {
+                    printOutput("Unable to parse date! Please try again.", true);
+                    continue;
+                }
+                taskList.add(d);
+                printOutput("Added new deadline: " + taskArgs[0] + System.lineSeparator() + "\tDue by: " + d.getDue());
+            } else if (userInput.toLowerCase().startsWith("event")) {
+                if (!userInput.toLowerCase().contains("/at")) {
+                    printOutput("usage: event <name of item> /at <dd-MMM-yyyy HH:mm:ss OR next [weekday]>", true);
+                    continue;
+                }
+                String[] taskArgs = userInput.substring(5).strip().split("/at");
+
+                Event e = new Event(taskArgs[0].strip(), taskArgs[1].strip());
+                if (e.getStart() == null) {
+                    printOutput("Unable to parse date! Please try again.", true);
+                    continue;
+                }
+                taskList.add(e);
+                printOutput("Added new event: " + taskArgs[0] + System.lineSeparator() + "\tStart at: " + e.getStart());
+            } else if (userInput.toLowerCase().startsWith("todo")) {
+                String taskString = userInput.substring(4).strip();
+                ToDo t = new ToDo(taskString);
+                taskList.add(t);
+                printOutput("Added new todo: " + t.toString());
+            } else {
+                printOutput("Commands: bye list todo deadline event", false);
+            }
+        }
+        printOutput("Bye. Hope to see you again soon!", true);
+    }
```

Comment:
The while loop is a little long and affects the ease of readability. Consider shortening the loop or separate some code into functions/classes.